tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsWriter.cs

```diff
@@ -238,7 +237,7 @@ private async Task ProcessQueueLoopAsync()
                 continue;
             }
 
-            _processingMutex.Wait();
+            await _processingMutex.WaitAsync().ConfigureAwait(false);
```

Comment:
So just to clarify, we sit on the thread pool for this function (compared to `SerializeTracesLoop`) because we start this function with `_processTask = Task.Run(ProcessQueueLoopAsync);` where the other one we do a `            _serializationTask = automaticFlush ? Task.Factory.StartNew(SerializeTracesLoop, TaskCreationOptions.LongRunning)`

Which gets it out of the thread pool.

Should we do the same for both and update that one to this as well?

Reply:
> Should we do the same for both and update that one to this as well?

As I understand it, that wouldn't work here because we're using `async`/`await`. Using `LongRunning` makes sense for the non-async method, as it (potentially) stays out of the pool entirely, but when you use async/await you lose the benefit seeing as you're going to be resuming on different threads potentially anyway.

cc @kevingosse to sense check me 😄